core/services/ocr2/plugins/directrequestocr/reporting.go

```diff
@@ -220,27 +241,60 @@ func (r *directRequestReporting) Report(ctx context.Context, ts types.ReportTime
 		return false, nil, err
 	}
 	return true, reportBytes, nil
-
 }
 
 // ShouldAcceptFinalizedReport() complies with ReportingPlugin
 func (r *directRequestReporting) ShouldAcceptFinalizedReport(ctx context.Context, ts types.ReportTimestamp, report types.Report) (bool, error) {
 	r.logger.Debug("directRequestReporting ShouldAcceptFinalizedReport phase", commontypes.LogFields{
-		"epoch": ts.Epoch,
-		"round": ts.Round,
+		"epoch":    ts.Epoch,
+		"round":    ts.Round,
+		"oracleID": r.genericConfig.OracleID,
+	})
+
+	// NOTE: The output of the Report() phase needs to be later decoded by the contract. So unfortunately we
+	// can't use anything more convenient like protobufs but we need to ABI-decode here instead.
+	decoded, err := r.reportCodec.DecodeReport(report)
+	if err != nil {
+		r.logger.Error("directRequestReporting unable to decode report built in reporting phase", commontypes.LogFields{"err": err})
+		return false, err
+	}
+
+	allIds := []string{}
+	needTransmissionIds := []string{}
+	for _, item := range decoded {
+		reqIdStr := formatRequestId(item.RequestID)
+		allIds = append(allIds, reqIdStr)
+		_, err := r.pluginORM.FindById(sliceToByte32(item.RequestID))
+		if err != nil {
+			r.logger.Warn("directRequestReporting request doesn't exist locally", commontypes.LogFields{"requestID": reqIdStr})
+			continue
```

Comment:
I am not sure this is a good idea.
The fact that the request does not exist locally, yet a report for this request is finalized might indicate that this oracle is slow and has not received the request yet.
This does not necessarily  create unnecessary transactions, it could be that only the transmitter of this stage has not received the request, but everyone else did.
Unless I misunderstood something.

Reply:
We discussed the tradeoffs offline. I'll document them in the reliability doc and we'll revisit this once we have some production data.

Reply:
Could you also add a comment mentioning that the observation plugin function should guarantee that there exist at least f+1 oracles who have the request locally, otherwise this violates the liveness of transmission?